### ISSUE TITLE
[strict-boolean-expressions] Add `ignore-rhs` option

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#755]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": "error"
  }
}
```

```TS
// OR
function valueOrDefault(a?: string) {
    return a || "the default";
}

// AND
const condition = true;
condition && valueOrDefault();
```

**Expected Result**

No error (with an option or options).

**Actual Result**

Error: `Unexpected non-boolean in conditional.` (twice)

**Additional Info**

TSLint has an `ignore-rhs` option (see [documentation](https://palantir.github.io/tslint/rules/strict-boolean-expressions/)).

It might make sense to also consider [this issue](https://github.com/palantir/tslint/issues/4671).

#698 would also be required (at least for the first example).

Other links:

* [TSLint orginal issue](https://github.com/palantir/tslint/issues/3279)
* [TSLint duplicate issue](https://github.com/palantir/tslint/issues/4158)
* [TSLint PR](https://github.com/palantir/tslint/pull/4159)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
#691

**Comment by JamesHenry** [comment]
The PR has been merged
